src/minichain/__main__.py

```
@@ -0,0 +1,1302 @@
+    print(f"[{timestamp}] [{tag}] {message}")
+
+
+def _acquire_node_pid_file(data_dir: Path) -> Path:
+    data_dir.mkdir(parents=True, exist_ok=True)
+    pid_file = _node_pid_path(data_dir)
+    existing_pid = _read_pid_file(pid_file)
+    if existing_pid is not None and _process_exists(existing_pid):
+        raise ValueError(
+            f"node already running for data_dir={data_dir} with pid={existing_pid}; "
+            "use `minichain --data-dir <dir> node stop` first"
+        )
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)